few/variation.py

```diff
@@ -25,7 +25,10 @@ def variation(self,parents):
                 offspring = copy.deepcopy(list(x for i,x in zip(self.ml.coef_, self.valid(parents)) if  (i != 0).any()))
             elif hasattr(self.ml,'feature_importances_'):
                 # for tree methods, filter our individuals with 0 feature importance
-                offspring = copy.deepcopy(list(x for i,x in zip(self.ml.feature_importances_, self.valid(parents)) if  i != 0))
```

Comment:
it looks like you just removed the old way of saving offspring. instead, you should use the self.weight_parents flag you introduced to check whether or not to execute the code below (i.e. weight the parents) instead of just removing parents with coefficients / feature importances of zero.